fbw-common/src/systems/instruments/src/EFB/Dispatch/Pages/LoadsheetPage.tsx

```diff
@@ -75,12 +87,72 @@ export const LoadSheetWidget = () => {
     setImageSize(cImageSize);
   };
 
+  function filterLoadsheetSection(section) {
+    const cleanStyles = (html) => {
+      return html
+        .replace(/^<div [^>]+>/, '')
+        .replace(/<\/div>$/, '')
+        .replace(/line-height:\s*\d+(\.\d+)?(px)?;?/g, '')
+        .replace(/font-size:\s*\d+px;?/g, '')
+        .replace(/font-family:\s*[^;]+;/g, '')
+        .replace(/<a\b[^>]*>[\s\S]*?<\/a>/g, '');
+    };
+
+    if (['Maps', 'Map', 'UAD Charts'].includes(section)) {
+      const anchorPattern = /<a\s+class="ofpmaplink"[^>]*href="([^"]+)"[^>]*>(.*?)<\/a>/g;
+      const filteredAnchors = [...loadsheet.matchAll(anchorPattern)].map((match) => ({
+        href: match[1],
+        innerHTML: match[2],
+      }));
+
+      return `
+          <pre>
+            ${filteredAnchors.map(({ href, innerHTML }) => `<a href="${href}" target="_blank">${innerHTML}</a>`).join('\n')}
+          </pre>
+        `;
+    }
+
+    if (section !== 'All') {
+      const currentIndex = loadsheetSections.indexOf(section);
+      let nextSection = currentIndex + 1 < loadsheetSections.length ? loadsheetSections[currentIndex + 1] : null;
+      if (['Maps', 'Map', 'UAD Charts'].includes(nextSection)) {
+        nextSection = currentIndex + 2 < loadsheetSections.length ? loadsheetSections[currentIndex + 2] : null;
+      }
+
+      const sectionMarkerPattern = new RegExp(
+        `<!--BKMK///${section}///\\d-->[\\s\\S]*?(?=<!--BKMK///${nextSection || ''}///\\d-->|$)`,
+        'g',
+      );
```

Comment:
Add some documentation to explain what this regex does in a comment

Reply:
The loadsheet received from Simbrief is in HTML format, for example:

You can see a couple of the section-markers (in green) in the OFP.
![image](https://github.com/user-attachments/assets/3d4d61a0-9432-47a9-9f2b-ed4c6b3538bf)

Using `[\\s\\S]*?` the regex captures all content, including line breaks, between two section markers
`<!--BKMK///${section}///\\d-->` (the selected section by the user) 
and 
`<!--BKMK///${nextSection || ''}///\\d-->` (next section, if it exists)

Reply:
You can actually just use the DOM API rather than trying to parse it as a string. Comment nodes are nodeType 8 https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType